lib/utils.js

```diff
@@ -0,0 +1,315 @@
+      exiting function.
+
+      @param call: an esprima node representing a function call
+      @param exits: an array of names of exiting functions
+
+      @return: a bool of whether the call is to exit or an exiting function
+   */
+   if (call.type === 'CallExpression'){
+      if (is_exit(call.callee, exits)){
+         return true;
+      }
+      var name = call.callee.name;
+      for (var i = 0; i < exits.length; i++) {
+         if (name === exits[i]){
+            return true;
+         }
+      }
+   }
+
+   return false;
+}
+
+function is_exit(node, function_names){
+   /*
+      Tests whether a node represents an exit call.
+
+      @param node: an esprima-parsed node
+      @param function_names: an optional array of names of functions which call 
+         exit in their body
+
+      @return: a bool of whether the node represents an exit call
+   */
+   var exits = ['exit'];
+   if (Array.isArray(function_names)){
+      exits = exits.concat(function_names);
+   }
+
+   switch(node.type){
+      case 'ExpressionStatement':
+         return handle_call(node.expression, exits);
+      case 'MemberExpression':
+         return handle_call(node.object, exits);
+      case 'VariableDeclaration':
+         for (var i = node.declarations.length - 1; i >= 0; i--) {
+            var init = node.declarations[i].init;
+            if (init && is_exit(init)){
+               return true;
+            }
+         }
+         break;
+      case 'CallExpression':
+         return handle_call(node, exits);
+      case 'ReturnStatement':
+         return is_exit(node.argument);
+      case 'Identifier':
+         for (var j = exits.length - 1; j >= 0; j--) {
+            var temp_exit = {
+               'type': 'Identifier',
+               'name': exits[j],
+            };
+            if (vars_equiv(node, temp_exit)){
+               return true;
+            }
+         }
+   }
+
+   return false;
+}
+
+function has_exit(node, function_names){
+   /*
+      Checks function bodies for an exit call.
+
+      @param node: an esprima node with `body` property or an array of esprima 
+         nodes
+      @param function_names: an array of names of functions that always exit
+
+      @return: a bool of whether `node` always exits
+   */
+   var actual_body = [];
+   if (node && node.body){
+      actual_body = node.body;
+      if (actual_body.body){
+         actual_body = actual_body.body;
+      }
+   }
+   for (var i = 0; i < actual_body.length; i++) {
+      if (is_exit(actual_body[i], function_names)){
+         return true;
+      }
+   }
+
+   return false;
+}
+
+function nodes_equal(node1, node2){
+   /* 
+      Checks whether two nodes are equal by comparing the range of code each node covers.
+
+      @param node1: an esprima node with range property
+      @param node2: an esprima node with range property
+
+      @return: a bool of whether the nodes are equal
+   */
+
+   // range is an array with two elements
+   if (node1.range){
```

Comment:
This could be all one line...

`return node1.range && node1.range[0] === node2.range[0] && node1.range[1] === node2.range[1]`